### PR TITLE
Provide more APIs for job constraint

### PR DESCRIPTION
**Pull request by WenboZhao**

### CODE REVIEW

File: jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/Constraints.java [hunk 1]
```diff
@@ -66,4 +66,26 @@ public static Constraint parseFrom(JSONArray constraint) {
...
         }
     }
+
+    /**
+     * @param constraint the string representation of a constraint. The string could be either the form
+     *                   "attribute,operator" or the form "attribute,operator,value1,value2,...".
+     *                   See {@link Constraint}.
+     * @return the parsed constraint.
+     */
+    public static Constraint parseFrom(String constraint) {
```
Comment (wyegelwel) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/Constraints.java]: What caused you to want to add this?

Comment (WenboZhao) [reply jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/Constraints.java]: See this https://github.com/twosigma/spark/pull/27/files

Comment (WenboZhao) [reply jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/Constraints.java]: I would like to make sure you are happy to add this API first. Then I will do some refactoring later

File: jobclient/src/test/java/com/twosigma/ConstraintTest.java [hunk 2]
```diff
@@ -30,4 +31,12 @@ public void testScope() {
         Assert.assertEquals(c.getOperator(), Operator.EQUALS);
         Assert.assertEquals(c.toJson().getString(2), "foo");
     }
+
+    @Test
+    public void testParseFrom() {
```
Comment (wyegelwel) [inline jobclient/src/test/java/com/twosigma/ConstraintTest.java]: This is a great test!

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java [hunk 3]
```diff
@@ -94,7 +94,8 @@ public static Status fromString(String name) {
         private JSONObject _container;
         private Map<String, String> _labels = new HashMap<>();
         private List<UUID> _groups = new ArrayList<>();
-        private Set<Constraint> _constraints = new HashSet<>();
+        // Use LinkedHashSet to ensure the insertion order will be kept.
```
Comment (wyegelwel) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: This is a helpful comment

**Pull request merged**